src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -649,6 +679,95 @@ def get_observation(self, event: Event) -> pd.DataFrame:
         for col in ["employer_address_id", "employer_name"]:
             df_w2[col] = df_w2["employer_id"].map(emp[col])
 
+        # re-initialize income-to-date series for next year of income counting
+        self.income_last_year = self.income_this_year
+        self.income_this_year = empty_income_series()
+
+        return df_w2[self.output_columns]
+
+
+class TaxDependentsObserver(BaseObserver):
+    """Class for observing columns relevant to identifying dependents in tax data.
+
+    This is most important for the 1040 data, but relies on data from
+    the W2 observer, and is better represented in a "long" form with a
+    row for each guardiant/dependent pair
+
+    """
+
+    INPUT_VALUES = ["household_details"]
+    ADDITIONAL_INPUT_COLUMNS = [
+        "alive",
+        "in_united_states",
+        "tracked",
+        "ssn",
+    ]
+    OUTPUT_COLUMNS = [
+        "guardian_id",
+        "dependent_id",
+        "first_name_id",
+        "middle_name_id",
+        "last_name_id",
+        "age",
+        "date_of_birth",
+        "sex",
+        "ssn",
+        "address_id",
+        "tax_year",
+    ]
+
+    def __init__(self, w2_observer):
+        super().__init__()
+        self.w2_observer = w2_observer
+
+    def __repr__(self):
+        return f"TaxDependentsObserver()"
+
+    @property
+    def name(self):
+        return f"tax_dependents_observer"
+
+    @property
+    def input_columns(self):
+        return self.DEFAULT_INPUT_COLUMNS + self.ADDITIONAL_INPUT_COLUMNS
+
+    @property
+    def input_values(self):
+        return self.INPUT_VALUES
+
+    @property
+    def output_columns(self):
+        return self.OUTPUT_COLUMNS
+
+    def setup(self, builder: Builder):
+        super().setup(builder)
+        self.clock = builder.time.clock()
+
+    def to_observe(self, event: Event) -> bool:
+        """Observe if Jan 1 falls during this time step"""
+        tax_date = dt.datetime(event.time.year, 1, 1)
```

Comment:
It's a little difficult to think through the way this observer works. As you've noted below in `get_observation`, the technically correct thing would be to observe these dependent relationships throughout the tax year. But even with the approach we've chosen here of observing at the end of the tax year, we will report `age` and `address_id` for dependents on the 1040 (in April) as of January 1st. Let's note this limitation somewhere.

Reply:
> It's a little difficult to think through the way this observer works. As you've noted below in get_observation, the technically correct thing would be to observe these dependent relationships throughout the tax year. But even with the approach we've chosen here of observing at the end of the tax year, we will report age and address_id for dependents on the 1040 (in April) as of January 1st. Let's note this limitation somewhere.

I was thinking that it was the dependent's age and address for the previous year that was relevant, did I misunderstand that?

Reply:
Ah, good question. I did not think about that.

All I can find in the docs is:

> Sampling on a single time step is not representative of how tax documents are compiled.

The observer as you have it here observes the filer's age and address as of April and the dependents' age and address as of January. It doesn't seem like those can both be right, but I'm not sure which one is the limitation! Maybe you can just add a note to that effect.

Reply:
Good idea, I will add a note in the observer's docstring